connector/keystone/keystone.go

```diff
@@ -215,24 +223,31 @@ func (p *conn) Refresh(
 	if err != nil {
 		return identity, fmt.Errorf("keystone: failed to obtain admin token: %v", err)
 	}
+
 	ok, err := p.checkIfUserExists(ctx, identity.UserID, token)
 	if err != nil {
 		return identity, err
 	}
 	if !ok {
 		return identity, fmt.Errorf("keystone: user %q does not exist", identity.UserID)
 	}
-	if scopes.Groups {
-		groups, err := p.getUserGroups(ctx, identity.UserID, token)
+
+	if p.groupsRequired(scopes.Groups) {
+		var data connectorData
+		if err := json.Unmarshal(identity.ConnectorData, &data); err != nil {
+			return identity, fmt.Errorf("github: unmarshal access token: %v", err)
```

Comment:
Does this approach affect other connectors like `github` as well ?

Reply:
Typo